jwst/flatfield/flat_field.py

```diff
@@ -375,32 +404,51 @@ def do_MOS_flat_field(output_model, flat_model, flat_suffix=None):
     else:
         interpolated_flats = None
 
+    any_updated = False
     for (k, slit) in enumerate(output_model.slits):
+        log.debug("Processing slit %s", slit.name)
         # Get the wavelength of each pixel in the extracted slit data.
         ysize, xsize = slit.data.shape
         grid = np.indices((ysize, xsize), dtype=np.float64)
         # The arguments are the X and Y pixel coordinates (in that order).
```

Comment:
The WCS works with full frame coordinates. X and Y should be converted to the image coordinates by shifting them by `slit.xstart-1` and `slit.ystart-1`. This will put them in image coordinates. If the image is a subarray, the x/y subarray shifts are absorbed in the WCS, there's no need to correct for this. We should discuss whether there are better options for the WCS. The reason it works this way now is in some peculiarities of the MultiSlitModel  which make it hard to be consistent (should discuss the MultiSlitModel too).